### PR TITLE
:sparkles: - Use htmx to async update messages.

### PR DESCRIPTION
**Pull request by svenvandescheur**

### CODE REVIEW

**Comment by annashamray** [comment]
Is it possible to unit test? (if not then should we really use it?)

**Comment by svenvandescheur** [comment]
> Is it possible to unit test? (if not then should we really use it?)

It uses the same logic as the previous implementation except async. The actual implementation is iirc 4 properties, testing those would just be retesting the htmx library.

**Comment by annashamray** [comment]
I still think this should be tested. If unit testing is not a good idea, end-to-end tests could be introduced

**Comment by svenvandescheur** [comment]
> I still think this should be tested. If unit testing is not a good idea, end-to-end tests could be introduced

Ok, let's see if I can add some wettest to. test the integration.

**Comment by svenvandescheur** [comment]
@annashamray I've added selenium for tests for this.

**Review by alextreme — APPROVED** [review]
Does seem like a bit of HTML magic, but lets try this out.

@annashamray see if you can help Sven with the requirements FIXME for adding selenium

**Comment by annashamray** [comment]
@alextreme dependencies didn't compile because there was a version conflict between our `certifi` and `selenium` `certifi`. I updated our version and it seems to work

File: src/open_inwoner/components/templates/components/Form/Form.html
Comment (annashamray) [inline src/open_inwoner/components/templates/components/Form/Form.html:13]: `action` variable is not documented in template tags

**Comment by alextreme** [comment]
Great thanks @annashamray !

File: src/open_inwoner/components/templates/components/Form/Form.html
Comment (annashamray) [inline src/open_inwoner/components/templates/components/Form/Form.html:15]: Should we also add polling for messages? According to the docs it can be done with 1 attribute
@alextreme @svenvandescheur what do you think?

File: src/open_inwoner/components/templates/components/Form/Form.html
Comment (annashamray) [inline src/open_inwoner/components/templates/components/Form/Form.html:16]: Why to use this tag? I commented it and nothing changed

File: src/open_inwoner/accounts/tests/test_inbox_page.py
Comment (annashamray) [inline src/open_inwoner/accounts/tests/test_inbox_page.py:126]: Nice! Perhaps in the future we can also test it for Chrome

**Review by annashamray — CHANGES_REQUESTED** [review]

File: src/open_inwoner/components/templates/components/Form/Form.html
Comment (annashamray) [reply src/open_inwoner/components/templates/components/Form/Form.html:15]: https://htmx.org/docs/#polling

File: .github/workflows/code-quality.yml
Comment (svenvandescheur) [reply .github/workflows/code-quality.yml:61]: I've added this line because sometimes the build seems to fail, maybe the runners are migrating to node. I preferably stick with a more recent (latest) node on the ci so we catch stuff early. Figuring out why/if we need legacy-peer-deps is IMO another task.

**Review by alextreme — APPROVED** [review]
Can be merged after Anna's approval

File: src/open_inwoner/components/templates/components/Messages/Messages.html
Comment (annashamray) [inline src/open_inwoner/components/templates/components/Messages/Messages.html:15]: Should we make it 5 seconds? @alextreme  ?

File: src/open_inwoner/components/templatetags/form_tags.py
Comment (annashamray) [inline src/open_inwoner/components/templatetags/form_tags.py:47]: I've just noticed that we already have `form_action` variable. Are they duplicates?

File: src/open_inwoner/components/templates/components/Messages/Messages.html
Comment (annashamray) [inline src/open_inwoner/components/templates/components/Messages/Messages.html:13]: Please add test for polling

**Review by annashamray — CHANGES_REQUESTED** [review]
Polling doesn't work on my local machine. Does it work for you?

**Comment by svenvandescheur** [comment]
> Polling doesn't work on my local machine. Does it work for you?

yes

**Review by Bartvaderkin — APPROVED** [review]

File: src/open_inwoner/components/templates/components/Form/Form.html
Comment (annashamray) [reply src/open_inwoner/components/templates/components/Form/Form.html:13]: `action` is not documented again

**Review by annashamray — CHANGES_REQUESTED** [review]
I have one remark about documentation which is not resolved 

Polling still doesn't work for me, but maybe the problem is in my config, so I'll approve after the template variable is documented

**Comment by annashamray** [comment]
@Bartvaderkin @svenvandescheur 
Polling works on your local machine right? How do you test it?

My steps:
1. log in as use A, go to "Mijn berichten" page with user B
2. in other session hijack user B, go to "Mijn berichten" and send message to user A
3. Wait for this message to appear in the user A session
The message doesn't appear

**Comment by svenvandescheur** [comment]
> @Bartvaderkin @svenvandescheur Polling works on your local machine right? How do you test it?
> 
> My steps:
> 
> 1. log in as use A, go to "Mijn berichten" page with user B
> 2. in other session hijack user B, go to "Mijn berichten" and send message to user A
> 3. Wait for this message to appear in the user A session
>    The message doesn't appear

I made screencapture:

oippolling

**Comment by annashamray** [comment]
Thanks! I'll test it on our test env after deployment and will try to understand why it doesn't work locally for me

**Review by annashamray — APPROVED** [review]

**Pull request merged**
